### ISSUE TITLE
Route and addContentTypeParser

### ISSUE THREAD

**Issue by Sn00z3r** [fastify/fastify#1499]
From what I can gather the addContentTypeParser totally ignores routes you have defined. So if the content type matches your parser it'll just run the function you defined in the addContentTypeParser.

Is there any way to get this working with fastify.route() or do I need to create my own route selector code for this specific content type?

Best regards!

**Comment by delvedor** [comment]
Hello! I'm sorry, I'm not sure I've understood the issue.
Can you rephrase it? A code example would help as well!
Thanks!

**Comment by Sn00z3r** [comment]
Hey,

Well, I have a post route in my application:

`    fastify.route({
        method: 'POST',
        url: '/voice/Error.xml',
        handler: async function (request, reply) {
            console.info("lalalla")
            reply.code(400)
            .send();
        }
      });`

Now when I send a post to this route with a text/plain content type I get unsupported content-type as a return code.

So I added 
`fastify.addContentTypeParser( 'text/plain', { parseAs: 'string' }, (req, body, done) => done("THIS IS A TEST!") )`

However now, whenever I send any request that uses text/plain, I always get the response THIS IS A TEST, all my code in the routes gets ignored.

**Comment by Sn00z3r** [comment]
Alright,

Ik was not on fastify 2.x, my apologies, all is well now :p

**Comment by Sn00z3r** [comment]
Alright,

Ik was not on fastify 2.x, my apologies, all is well now :p
